src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,401 @@
+                        getFullClassNameFromClassFilePath(
+                            manualFilePathToTestFilePath(classFile.toString())),
+                        manualFilePathToTestFilePath(classFile.toString())
+                    );
+                    objectList.add(testClassesDTO);
+                }
+            }
+            Object[] arr = new Object[]{objectList};
+            testParameters.add(arr);
+        }
+        return testParameters;
+    }
+
+    /**
+     * Run tests for every TestClassesDTO object (in other words for every
+     * class that was generated by POG (every class of the test case)).
+     * Number of iterations depends on number of classes to test in this particular test case.
+     */
+    @Test
+    public void runForPackage() throws Exception {
+        caseClassesList = sortClassesList(caseClassesList);
+        for (TestClassesDTO classDTO : caseClassesList) {
+            currentClassesDTO = classDTO;
+            runTestCase();
+        }
+    }
+
+    /**
+     * Compiles two classes (manual and generated by POG)
+     *
+     * @return CompilationDTO object with Class objects (for manual and generated by POG
+     * test classes) and result of compilation
+     */
+    private CompilationResultDTO prepareAndCompileTestClassesPair() throws Exception {
+        PageObjectsGenerator pog = initPog(
+            currentClassesDTO.getJsonPath(),
+            currentClassesDTO.getTestURL(),
+            currentClassesDTO.isCheckLocatorUniqueness(),
+            currentClassesDTO.isForceGenerateFiles());
+
+        TestClassesCompiler compiler = new TestClassesCompiler(pog);
+        CompilationResultDTO compilationResult = compiler.compileClasses(
+            currentClassesDTO.getTestClassFilePath(),
+            currentClassesDTO.getManualClassFilePath(),
+            currentClassesDTO.getTestClassFullName(),
+            currentClassesDTO.getManualClassFullName()
+        );
+        assertEquals(true, compilationResult.isCompilationSuccess());
+        return compilationResult;
+    }
+
+    /**
+     * Run all test methods for one generated class (and compare it to manual class)
+     */
+    private void runTestCase() throws Exception {
+        CompilationResultDTO compilationResult = prepareAndCompileTestClassesPair();
+        currentClassesDTO.setTestClass(compilationResult.getExpectedClass());
+        currentClassesDTO.setManualClass(compilationResult.getManualClass());
+
+        testNameOfClass();
+        testAnnotationOfClass();
+        testModifiersOfClass();
+        testSuperClassOfClass();
+        testFields();
+    }
+
+    /**
+     * Compare names of test and manual classes. For example:
+     * test.page.Site           - test class
+     * manual.google.page.Site  - manual class
+     * will pass assertEquals
+     */
+    private void testNameOfClass() throws Exception {
+        String[] testClassName = currentClassesDTO
```

Comment:
assertEquals("File names do not match", FilenameUtils.getName(expectedClassName)
        FilenameUtils.getName(actualClassName));

Note that expected value goes first in assertion